src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -281,6 +297,76 @@ public bool IsSolutionOpen
             }
         }
 
+        public async Task<Tuple<bool, bool>> CheckSolutionUIPreConditionsAsync()
```

Comment:
And about the performance, then it's hardly an overhead given it's only used once and disposed immediately, and top of that just have two boolean. if you compare out vs tuple for 100000 times for a simple code like this, then you might see 1ns difference.

And on top of all this, `out` can't be used here since this is a async method which doesn't allow out or ref.